core/arch/arm/kernel/generic_boot.c

```diff
@@ -302,15 +302,15 @@ static void init_runtime(unsigned long pageable_part)
 	 */
 	mm = tee_mm_alloc2(&tee_mm_vcore, tee_mm_vcore.lo,
 			(vaddr_t)(__text_init_start - tee_mm_vcore.lo));
-	TEE_ASSERT(mm);
+	panic_unless(mm);
```

Comment:
As all other tee_mm allocations are use to access data, core will crash for shure, but this alloc is never used, hence it should be tested. But i agree the resulting C code sequence is somewhat strange.